File: BE/project/src/main/java/com/ssafy/project/domain/Letter.java
package com.ssafy.project.domain;


import com.ssafy.project.dto.LetterDto;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;

@Entity
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Table(name = "letter")
@EntityListeners(AuditingEntityListener.class)
public class Letter {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "letter_id")
    private Long letterId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "child_id")
    private Child child;

    private String letter;

    private String reply;

    @Column(name = "book_title")
    private String bookTitle;

    private String role;

    private String letterFileName;

    @CreatedDate
    private LocalDateTime createdAt;


    public static LetterDto entityToDto(Letter letter) {
        return LetterDto.builder()
                .bookTitle(letter.bookTitle)
                .role(letter.role)
                .content(letter.letter)
                .letterFileName(letter.letterFileName)
                .reply(letter.reply)
                .createdAt(String.valueOf(letter.createdAt))
                .build();
    }

}

File: FE/src/types/letter.ts
export interface LetterInfo {
    bookTitle: string; // 책 이름
    role: string; // 답장할 인물
    childName: string; // 편지를 보내는 어린이
    content: string; // 편지 내용
    letterFileName: string; // 음성 파일 S3
    letterUrl: string; // 음성 파일 S3
    reply: string; // 등장인물 답변
    createdAt: string; // 편지 전송 시간
}

export interface LetterData extends LetterInfo {
    bookPath: string;
}

export interface GPTRequest {
    fairyTale: string;
    role: string;
    childName: string;
    content: string;
    letterRecord?: string;
}
